src/mon/MgrMonitor.cc

```diff
@@ -895,6 +900,16 @@ bool MgrMonitor::prepare_command(MonOpRequestRef op)
       r = -EINVAL;
       goto out;
     }
+    if (!pending_map.any_support_module(module)) {
+      ss << "module " << module << " does not exist";
```

Comment:
I think we should move the check inside the `!pending_map.module_enabled` block, so that the test can still disable a non-existent module if the module was enabled with --force at some point.

Then the test should also be updated on the `foodnebizbangbash` line to expect that to fail.

Reply:
@jcsp how about set the retval =0, if pending_map.any_support_module(module) is false?

Reply:
Actually what I just suggested was wrong because it wouldn't be idempotent.  We need to return 0 from all these cases.

However, in the `!pending_map.any_support_module(module)` we can't just return, because it might still need removing from the list of enabled modules if it was added with --force.  We can do that by moving this block after the `!pending_map.module_enabled(module)` one, and just set the `ss` output rather then returning.

So I *think* the correct code would be:

```
    if (!pending_map.module_enabled(module)) {
      ss << "module " << module << " is already disabled";
      r = 0;
      goto out;
    }
    if (!pending_map.any_support_module(module)) {
      ss << "module " << module << " does not exist";
    }
     pending_map.modules.erase(module);
```

Reply:
@jcsp @tchaikov , updated. please take a look.